modules/howtos/examples/transactions-example.php

```diff
@@ -0,0 +1,489 @@
+      $content = $doc->content();
+      $newContent = array_merge(
+        ["transactions" => "are awesome"],
+        $content);
+      
+      $ctx->replace($doc, $newContent);
+  });
+  // end::replace[]
+}
+
+function removeExample($cluster, $collection) {
+  // tag::remove[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection) {
+      $doc = $ctx->get($collection, "docId");
+      $ctx->remove($doc);
+  });
+  // end::remove[]
+}
+
+function insertExample($cluster, $collection) {
+  // tag::insert[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection) { 
+      $adoc = $ctx->insert($collection, "docId", []);
+  });
+  // end::insert[]
+}
+
+function queryExamples($cluster) {
+  // tag::queryExamplesSelect[]
+  $inventory = $cluster->bucket('travel-sample')->scope('inventory');
+
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) { 
+      $st = "SELECT * FROM `travel-sample`.inventory.hotel WHERE country = $1";
+      $qr = $ctx->query(
+        $st,
+        TransactionQueryOptions::build()
+          ->positionalParameters(["United Kingdom"]));
+
+      foreach ($qr->rows() as $row) {
+        // do something
+      }
+    }
+  );
+  // end::queryExamplesSelect[]
+
+  // tag::queryExamplesUpdate[]
+  $hotelChain = 'http://marriot%';
+  $country = 'United States';
+  
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($hotelChain, $country) { 
+      
+      // TODO: scope parameter doesn't work
+      // $st = hotel SET price = $1 WHERE url LIKE $2 AND country = $3";
+      $st = "UPDATE `travel-sample`.inventory.hotel SET price = $1 WHERE url LIKE $2 AND country = $3";
+      $qr = $ctx->query(
+        $st,
+        TransactionQueryOptions::build()
+          ->positionalParameters([99.99, $hotelChain, $country]));
+          // ->scopeQualifier("`travel-sample`.`inventory`"));
+
+      print_r($qr->metaData()->metrics());
+      if ($qr->metaData()->metrics()["mutationCount"] != 2) {
+        // throw new \Exception("Mutation count not the expected amount.");
+        echo "WARN: Mutation count not the expected amount.\n";
+      }
+    }
+  );
+  // end::queryExamplesUpdate[]
+
+  // tag::queryExamplesComplex[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($hotelChain, $country) { 
+      // Find all hotels of the chain
+      $qr = $ctx->query(
+        'SELECT reviews FROM `travel-sample`.inventory.hotel WHERE url LIKE $1 AND country = $2',
+        TransactionQueryOptions::build()
+          ->positionalParameters([$hotelChain, $country]));
+          // ->scopeQualifier("`travel-sample`.`inventory`"));
+
+      // This function (not provided here) will use a trained machine learning model to provide a
+      // suitable price based on recent customer reviews.
+      function priceFromRecentReviews(Couchbase\QueryResult $qr) {
```

Comment:
Should be a TransactionQueryResult

Reply:
`Fatal error: Uncaught TypeError: priceFromRecentReviews(): Argument #1 ($qr) must be of type Couchbase\TransactionQueryResult, Couchbase\QueryResult given`

One for @avsej ?

Reply:
Yep this looks like an API bug.